pyomo/core/base/constraint.py

```diff
@@ -311,7 +311,7 @@ class _GeneralConstraintData(_ConstraintData):
         _active         A boolean that indicates whether this data is active
     """
 
-    __slots__ = ('_body', '_lower', '_upper', '_equality')
+    __slots__ = ('_body', '_lower', '_upper', '_equality', '_canonical_form_has_been_set', '_canonical_form')
```

Comment:
We want to ensure that if the model has been finalised, this is persisted if it is serialised.